src/signal_manager.sh

```diff
@@ -0,0 +1,66 @@
+include "$KW_LIB_DIR/kwio.sh"
+
+function get_valid_signals()
+{
+  local -r signal_list="$(trap -l | grep -oE 'SIG\S+')"
+  declare -gA SIGNAL_MANAGER_VALID_SIGNALS
+
+  for s in $signal_list; do
+    SIGNAL_MANAGER_VALID_SIGNALS["$s"]='1'
+  done
+}
+
+get_valid_signals
+
+# This functions prints the default message when interrupting kw
+function default_interrupt_handler()
+{
+  say $'\n'Oh no! An interruption! See ya...
+}
+
+# This function adds a new signal handler to an arbitrary signal
+#
+# @command The handler to be called when a signal is catched. This can
+#          can be a compound command (e.g. a function with parameters)
+#          but it must be passed as a single word (i.e. between quotes)
+# @signals The list of signals to attach to this handler
+#
+# If command is empty, use the default handler
+# If the signal is empty, use the default signals
+
+function signal_manager()
+{
+  local command="$1"
+  shift
+  local -a signals=("$@")
+
+  [[ -z "$command" ]] && command='default_interrupt_handler'
+
+  if [[ "${#signals}" == 0 ]]; then
+    signals=(SIGINT SIGTERM)
+  else
+    for s in "${signals[@]}"; do
+      if [[ ! -v SIGNAL_MANAGER_VALID_SIGNALS["$s"] &&
+        ! -v SIGNAL_MANAGER_VALID_SIGNALS[SIG"$s"] ]]; then
+        return 22 # EINVAL
+      fi
+    done
+  fi
+
+  trap "$command" "${signals[@]}" 2> /dev/null
```

Comment:
I think shellcheck has a good point here... Maybe you should add a check if the "$command" is something that exists, and after that, you can add `# shellcheck disable=SC2064` since we know that command is valid and it is a string that points to a function.
P.s.: Explain why you added this in the commit message.